src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -26,52 +26,70 @@ public static class ScriptJwtBearerExtensions
-                                },
-                                OnTokenValidated = c =>
-                                {
-                                    c.Principal.AddIdentity(new ClaimsIdentity(new Claim[]
-                                    {
-                                        new Claim(SecurityConstants.AuthLevelClaimType, AuthorizationLevel.Admin.ToString())
-                                    }));
+                        return Task.CompletedTask;
+                    },
+                    OnTokenValidated = c =>
+                    {
+                        c.Principal.AddIdentity(new ClaimsIdentity(new Claim[]
+                        {
+                            new Claim(SecurityConstants.AuthLevelClaimType, AuthorizationLevel.Admin.ToString())
+                        }));
 
-                                    c.Success();
+                        c.Success();
 
-                                    return Task.CompletedTask;
-                                }
-                            };
+                        return Task.CompletedTask;
+                    }
+                };
 
-                            o.TokenValidationParameters = CreateTokenValidationParameters();
+                o.TokenValidationParameters = CreateTokenValidationParameters(GetValidAudiences());
+
+                // TODO: DI (FACAVAL) Remove this once the work above is completed.
+                if (!SystemEnvironment.Instance.IsPlaceholderModeEnabled())
+                {
+                    // We're not in standby mode, so flag as specialized
+                    _specialized = 1;
+                }
+            });
+
+        private static string[] GetValidAudiences()
+        {
+            if (SystemEnvironment.Instance.IsPlaceholderModeEnabled() &&
+                SystemEnvironment.Instance.IsLinuxConsumptionOnLegion())
+            {
+                return new string[]
+                {
+                    ScriptSettingsManager.Instance.GetSetting(WebsitePodName)
+                };
+            }
 
-                            // TODO: DI (FACAVAL) Remove this once the work above is completed.
-                            if (!SystemEnvironment.Instance.IsPlaceholderModeEnabled())
-                            {
-                                // We're not in standby mode, so flag as specialized
-                                _specialized = 1;
-                            }
-                        });
+            return new string[]
+            {
+                string.Format(SiteAzureFunctionsUriFormat, ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName)),
+                string.Format(SiteUriFormat, ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName))
+            };
+        }
 
-        private static TokenValidationParameters CreateTokenValidationParameters()
+        private static TokenValidationParameters CreateTokenValidationParameters(string[] validAudiences)
```

Comment:
Also, it'd be ideal if you added a new unit test suite ScriptJwtBearerExtensionsTests to test this static method (make it internal for testing). That'd let you test this new conditional specialization logic. Looks like there wasn't an existing suite for you to add tests to, because previously the logic was pretty static.

Reply:
Made this public and added some tests.